### PR TITLE
Use `ArgsUsage` to be more helpful on arguments

### LINKED ISSUE #92: --help pages aren't helpful for subcommand arguments
```
% ./zypper-docker --help                                                   
NAME:
   zypper-docker - Patching Docker images safely

USAGE:
   zypper-docker [global options] command [command options] [arguments...]
   
VERSION:
   1.1.2
   
COMMANDS:
   images                       List all the images based on either OpenSUSE or SLES
   list-updates, lu             List all the available updates
   list-updates-container, luc  List all the available updates for the given container
   update, up                   Install the available updates
   list-patches, lp             List all the available patches
   list-patches-container, lpc  List all the available patches for the given container
   patch                        Install the available patches
   patch-check, pchk            Check for patches
   patch-check-container, pchkc Check for patches available for the given container
   ps                           List all the containers that are outdated
   help, h                      Shows a list of commands or help for one command
   
GLOBAL OPTIONS:
   --no-gpg-checks                                      Ignore GPG check failures and continue
   --gpg-auto-import-keys                               If new repository signing key is found, do not ask what to do; trust and import it automatically
   -f, --force                                          Ignore all the local caches
   -d, --debug                                          Show all the logged messages on stdout
   --add-host [--add-host option --add-host option]     Add a custom host-to-IP mapping (host:ip)
   --help, -h                                           show help
   --version, -v                                        print the version
   
% ./zypper-docker update --help     
NAME:
   zypper-docker update - Install the available updates

USAGE:
   zypper-docker update [command options] [arguments...]

OPTIONS:
   -l, --auto-agree-with-licenses       Automatically say yes to third party license confirmation prompt. By using this option, you choose to agree with licenses of all third-party software this command will install.
   --no-recommends                      By default, zypper installs also packages recommended by the requested ones. This option causes the recommended packages to be ignored and only the required ones to be installed.
   --replacefiles                       Install the packages even if they replace files from other, already installed, packages. Default is to treat file conflicts as an error.
   --author "cyphar"                    Commit author to associate with the new layer (e.g., "John Doe <john.doe@example.com>")
   --message "[zypper-docker] update"   Commit message to associated with the new layer
   
% ./zypper-docker patch-check --help
NAME:
   zypper-docker patch-check - Check for patches

USAGE:
   zypper-docker patch-check [arguments...]
```

None of those `[arguments...]` is actually helpful to someone using zypper-docker for the first time.

### PR DESCRIPTION
**Pull request by mssola**
Fixes #92

Signed-off-by: Miquel Sabaté Solà <msabate@suse.com>

### CODE REVIEW

**Comment by mssola** [comment]
Unfortunately, as you can see, there's no way to tell `cli` that there are no further arguments, so you have to provide a string like "<no arguments>".

**Comment by monstermunchkin** [comment]
An empty `ArgsUsage` would still print `[arguments...]`, right? Have you tried using `UsageText` which overrides the USAGE section? https://github.com/codegangsta/cli/pull/289

File: flags.go [hunk 1]
```diff
@@ -86,27 +86,45 @@ func newApp() *cli.App {
 	}
 	app.Commands = []cli.Command{
 		{
-			Name:   "images",
-			Usage:  "List all the images based on either OpenSUSE or SLES",
-			Action: getCmd("images", imagesCmd),
+			Name:      "images",
+			Usage:     "List all the images based on either OpenSUSE or SLES",
+			Action:    getCmd("images", imagesCmd),
+			ArgsUsage: "<no arguments>",
```
Comment (tboerger) [inline flags.go]: `ArgsUsage: " "` will do the trick :)

File: flags.go [hunk 2]
```diff
@@ -257,17 +292,26 @@ func newApp() *cli.App {
...
+to be used.`,
 		},
 		{
-			Name:   "ps",
-			Usage:  "List all the containers that are outdated",
-			Action: getCmd("ps", psCmd),
+			Name:      "ps",
+			Usage:     "List all the containers that are outdated",
+			Action:    getCmd("ps", psCmd),
+			ArgsUsage: "<no arguments>",
```
Comment (tboerger) [inline flags.go]: `ArgsUsage: " "` will do the trick :)

Comment (mssola) [inline flags.go]: Argh, why didn't I think about that ! :)

**Comment by mssola** [comment]
@monstermunchkin that indeed looks interesting but it's a bit of an overkill if this is the only problem. So, I'm taking @tboerger's approach :)

**Comment by tboerger** [comment]
LGTM

**Comment by monstermunchkin** [comment]
LGTM

**Comment by cyphar** [comment]
LGTM. My only concern is that instead of dynamically generated documentation, we use static documentation. This is a limitation of the `cli` package we're using, but still ... I don't like the fact that we can't have it dynamically generated (like in `python` with `argparse`).

**Pull request merged**